backend/packages/Upgrade/src/api/controllers/ExperimentClientController.v4.ts

```diff
@@ -0,0 +1,938 @@
+import {
+  JsonController,
+  Post,
+  Body,
+  UseBefore,
+  Get,
+  BodyParam,
+  Req,
+  InternalServerError,
+  Delete,
+  Patch,
+  OnUndefined,
+} from 'routing-controllers';
+import { ExperimentService } from '../services/ExperimentService';
+import { ExperimentAssignmentService } from '../services/ExperimentAssignmentService';
+import { ExperimentAssignmentValidator } from './validators/ExperimentAssignmentValidator';
+import { ExperimentUser } from '../models/ExperimentUser';
+import { ExperimentUserService } from '../services/ExperimentUserService';
+import { UpdateWorkingGroupValidator } from './validators/UpdateWorkingGroupValidator';
+import {
+  IExperimentAssignmentv4,
+  ISingleMetric,
+  IGroupMetric,
+  SERVER_ERROR,
+  IGroupMembership,
+  IUserAliases,
+  IWorkingGroup,
+  PAYLOAD_TYPE,
+  EXPERIMENT_TYPE,
+} from 'upgrade_types';
+import { FailedParamsValidator } from './validators/FailedParamsValidator.v1';
+import { ExperimentError } from '../models/ExperimentError';
+import { FeatureFlag } from '../models/FeatureFlag';
+import { FeatureFlagService } from '../services/FeatureFlagService';
+import { ClientLibMiddleware } from '../middlewares/ClientLibMiddleware';
+import { LogValidator } from './validators/LogValidator';
+import { MetricService } from '../services/MetricService';
+import { ExperimentUserAliasesValidator } from './validators/ExperimentUserAliasesValidator';
+import { Metric } from '../models/Metric';
+import * as express from 'express';
+import { AppRequest } from '../../types';
+import { env } from '../../env';
+import { MonitoredDecisionPointLog } from '../models/MonitoredDecisionPointLog';
+import { MarkExperimentValidatorv4 } from './validators/MarkExperimentValidator.v4';
+
+interface ILog {
```

Comment:
Is there any reason why the 3 versioned ExperimentClientController files have interfaces defined in the controller file instead of in `Types`? I can understand if they were private and local just to this file, but these look like they are effectively the public types for response DTOs of these endpoints. Unless I'm missing something, that's important type information that is getting buried because it's being treated as a private, local type when they could be a public, global type.

Can these be moved to types files so that they organized and exported like the other response types so that it is documentable and also not duplicated across files?